lib/clone.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+var Cesium = require('cesium');
+var ArrayStorage = require('./ArrayStorage');
+
+var defaultValue = Cesium.defaultValue;
+
+module.exports = clone;
+
+/**
+ * Clones an object, returning a new object containing the same properties.
+ * Modified from Cesium.clone to support typed arrays, buffers, and the ArrayStorage class.
+ *
+ * @param {Object} object The object to clone.
+ * @param {Boolean} [deep=false] If true, all properties will be deep cloned recursively.
+ * @returns {Object} The cloned object.
+ *
+ * @private
+ */
+function clone(object, deep) {
```

Comment:
I just looked at the tests it's being used in, I don't think it's needed and we should delete it and remove clone usage.  I'll comment specifically in the tests it's used with more details.